[MHV-45118] Updated Reply Draft Navigation

vsaleem opened the pull request:
## Summary

- When a user navigates away from a Reply Compose page, they will receive a navigation warning. 
- The updated navigation warning modal will display on Reply Draft screen when a navigation event occurs if a user attempts to navigate away from the page before saving their changes within message body.

- The expected behavior: When a user replies to a message, then user types text within  the message body text box.  User attempts to navigate away before saving the draft or before the draft is auto saved, then the user is presented a warning that their changes will be lost and given the option to continue editing the draft or to delete the draft.


## Related issue(s)
### [MHV-45118](https://jira.devops.va.gov/browse/MHV-45118) - Updated Reply Draft Navigation ###

> 
> User Story: As a Medical Records Developer, I want to Understand PHR refresh timing and when to call it, so that I can develop code using the PHR refresh functionality.
> 
> Acceptance Criteria:
> AC1 When a user makes text change and attempts to navigate away before a message is saved, they should receive a modal warning that their changes will be lost.
>
>AC2 When a user doesn’t make a text change, then attempts to navigate away, then they should navigation away successfully without modal.
>
>AC3 When a user makes a text change, and waits 10 sec., then autoSave should run
>
>AC4 When a user makes a text change and attempts to navigate away before saving the message, they should receive a modal warning that their changes will be lost and auto save should NOT run
>
>AC5 When a user attaches a file then attempts to navigate away, they should receive a modal. And when a user removes the attachment, without making any other text changes, then the user should be able to navigate away successfully without a modal.
>
>Links to UCD: N/A
## Testing done
- Updated Cy tests to capture navigation event changes -- additional tests are pending per the automation team

- _Describe what the old behavior was prior to the change:  After entering some text within the message body, and the user clicks the navigation menu (Inbox), there is no warning and the user is navigated to another screen._ 

## Screenshots
N/A


## What areas of the site does it impact?

*(This code was created for Secure Messaging, vets-website / Va.gov - Secure Messaging)*

## Acceptance criteria

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _Additional automation testing is requested_

Commit:
[MHV-45118] Updated Reply Draft Navigation

-Added effect on messageBody if value is present, to display navigationError alert

fazilqa committed:
skipped navigate-away-from-replying-message / will be refactored later

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/tests/e2e/secure-messaging-navigate-away-from-replying-message.cypress.spec.js:
```diff
@@ -2,7 +2,8 @@ import SecureMessagingSite from './sm_site/SecureMessagingSite';
 
-describe('verify signature', () => {
+// test have to be refactored after according to new scenario (modal warning should appear)
+describe.skip('verify signature', () => {
```
@vsaleem please see if @fazilqa may assist with refactoring in this same PR. Otherwise, I can try to assist

vsaleem replied:
@oleksii-morgun-va, @fazilqa plans to update this test and refactor the other existing tests once these changes make it in to staging.  QA has also completed testing on this as of 10/5.

Commit:
[MHV-45118] Updated Navigation RouteGuard

- fixed bug so that autoSave only initiates when a the RouteGuard modal is not present

- fixed bug so that if a user doesn't make a text change or attach file then they can't navigate away without a modal

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
[MHV-45118] Bug Fix - ComposeForm

- Added update so that autoSave will not run while the RouteLeavingGuard modal is visible on the ComposeForm

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
[MHV-45118] Updated Navigating away on ReplyForm

- User can navigate away when the messageBody is blank and hasn't been altered by text

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

fazilqa committed:
Update secure-messaging-compose-recipients-dropdown.cypress.spec.js

update selector

Commit:
Merge branch '45118-Navigating-Away-from-Reply-has-no-warning-mhv' of https://github.com/department-of-veterans-affairs/vets-website into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Commit:
[MHV-45118] Corrected linting err on cy test

Commit:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Asterix22 committed:
Merge branch 'main' into 45118-Navigating-Away-from-Reply-has-no-warning-mhv

Asterix22 committed:
Update secure-messaging-keyboard-nav-to-attachments-compose-page.cypress.spec.js

Asterix22 committed:
cypress fixes

alex-morgun-1 reviewed (approved):
approved.

KyleCardwell reviewed (approved):
Approved
